SimpleMortarMapperProcess fails Windows compilation

I got this error recently when compiling Kratos on Windows

1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,4,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$03V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<3,3,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$02$02V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,1>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,1>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$00@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,1>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$00@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<class Kratos::array_1d<double,3> >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@V?$array_1d@N$02@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::VariableComponent<class Kratos::VectorComponentAdaptor<class Kratos::array_1d<double,3> > >,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$VariableComponent@V?$VectorComponentAdaptor@V?$array_1d@N$02@Kratos@@@Kratos@@@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,0>::ExecuteExplicitMapping(void)" (?ExecuteExplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)
1>add_processes_to_python.obj : error LNK2019: unresolved external symbol "private: void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,0>::ExecuteImplicitMapping(void)" (?ExecuteImplicitMapping@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$0A@@Kratos@@AEAAXXZ) referenced in function "public: virtual void __cdecl Kratos::SimpleMortarMapperProcess<2,2,class Kratos::Variable<double>,0>::Execute(void)" (?Execute@?$SimpleMortarMapperProcess@$01$01V?$Variable@N@Kratos@@$0A@@Kratos@@UEAAXXZ)

@roigcarlo do you also have the same problem?

--- a/kratos/processes/simple_mortar_mapper_process.h
+++ b/kratos/processes/simple_mortar_mapper_process.h
@@ -50,7 +50,7 @@ namespace Kratos
 ///@{
     
 template< int TDim, int TNumNodes, class TVarType, HistoricalValues THist> 
-class SimpleMortarMapperProcess
+class KRATOS_API(KRATOS_CORE) SimpleMortarMapperProcess
         : public Process
 {
 public:

--- a/kratos/python/add_processes_to_python.cpp
+++ b/kratos/python/add_processes_to_python.cpp
@@ -289,14 +289,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<2, 2, Variable<double>, Historical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<2, 2, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NComponentHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<2, 2, component_type, Historical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<2, 2, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NComponentHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<2, 2, component_type, Historical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<2, 2, Variable<array_1d<double,3> >, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NVectorHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())
@@ -316,14 +316,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<2, 2, Variable<double>, NonHistorical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<2, 2, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NComponentNonHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<2, 2, component_type, NonHistorical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<2, 2, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NComponentNonHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<2, 2, component_type, NonHistorical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<2, 2, Variable<array_1d<double,3> >, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess2D2NVectorNonHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())
@@ -344,14 +344,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<3, 3, Variable<double>, Historical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<3, 3, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NComponentHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<3, 3, component_type, Historical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<3, 3, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NComponentHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<3, 3, component_type, Historical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<3, 3, Variable<array_1d<double,3> >, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NVectorHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())
@@ -371,14 +371,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<3, 3, Variable<double>, NonHistorical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<3, 3, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NComponentNonHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<3, 3, component_type, NonHistorical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<3, 3, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NComponentNonHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<3, 3, component_type, NonHistorical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<3, 3, Variable<array_1d<double,3> >, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D3NVectorNonHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())
@@ -399,14 +399,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<3, 4, Variable<double>, Historical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<3, 4, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NComponentHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<3, 4, component_type, Historical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<3, 4, component_type, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NComponentHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<3, 4, component_type, Historical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<3, 4, Variable<array_1d<double,3> >, Historical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NVectorHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())
@@ -426,14 +426,14 @@ void  AddProcessesToPython()
     .def("Execute",&SimpleMortarMapperProcess<3, 4, Variable<double>, NonHistorical>::Execute)
     ;
 
-    class_<SimpleMortarMapperProcess<3, 4, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NComponentNonHistorical", init<ModelPart&, component_type&>())
-    .def(init<ModelPart&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def(init<ModelPart&, component_type&, component_type&>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters>())
-    .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
-    .def("Execute",&SimpleMortarMapperProcess<3, 4, component_type, NonHistorical>::Execute)
-    ;
+//     class_<SimpleMortarMapperProcess<3, 4, component_type, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NComponentNonHistorical", init<ModelPart&, component_type&>())
+//     .def(init<ModelPart&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def(init<ModelPart&, component_type&, component_type&>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters>())
+//     .def(init<ModelPart&, component_type&, component_type&, Parameters, LinearSolverType::Pointer>())
+//     .def("Execute",&SimpleMortarMapperProcess<3, 4, component_type, NonHistorical>::Execute)
+//     ;
 
     class_<SimpleMortarMapperProcess<3, 4, Variable<array_1d<double,3> >, NonHistorical>, bases<Process>, boost::noncopyable >("SimpleMortarMapperProcess3D4NVectorNonHistorical", init<ModelPart&, Variable<array_1d<double,3> >&>())
     .def(init<ModelPart&, Variable<array_1d<double,3> >&, Parameters>())

--- a/kratos/utilities/exact_mortar_segmentation_utility.h
+++ b/kratos/utilities/exact_mortar_segmentation_utility.h
@@ -74,7 +74,7 @@ namespace Kratos
  * This utility calculates the exact integration necessary for the Mortar Conditions
  */
 template< unsigned int TDim, unsigned int TNumNodes, bool TBelong = false>
-class ExactMortarIntegrationUtility
+class KRATOS_API(KRATOS_CORE) ExactMortarIntegrationUtility
 {
 public:
     ///@name Type Definitions

